transcom/mymove
# [MB-14747] [MB-14709] Update customer and office move APIs to save ppm closeout location

## [Jira ticket](https://dp3.atlassian.net/browse/MB-14747) for this change

## Summary

This modifies one existing customer API `PATCH /moves/{id}` and creates a new office API `PATCH /moves/{locator}/closeout-office` to save the chosen closeout office from each app.

For the customer endpoint I had to remove an unused field `selectedMoveType` that is no longer being utilized and added the closeoutOfficeID to the payload.  The endpoint also wasn't using eTag's so I added that header to the request.  I moved the actual database logic into the shared service with the office. 

## Setup to Run Your Code

<summary>💻 You will need to use two separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

##### Terminal 2

Start the Go server locally.

```sh
make server_run
```

### Additional steps



Testing the customer side update move endpoint
1. Login to the customer app milmovelocal:3000 with an account that has created orders
2. Grab the CSRF value from the cookie and the eTag from the move in one of the responses (logged_in request) 
3. Send an API request using the tool of your choice (Postman, Swagger-UI, Devtools)
4. Set the CSRF-Token and If-Match headers
5. Add a JSON body setting the closeoutOfficeId property to a transportation office id in your local database that provides closeout.
6. See that the response includes the closeout office id and nested closeout office data

Testing the office side update move endpoint
1. Login to the office app officelocal:3000 as the services counselor user
2. Grab the CSRF value from the cookie and the eTag from the move in one of the responses (/moves/{locator} request) 
3. Send an API request using the tool of your choice (Postman, Swagger-UI, Devtools)
4. Set the CSRF-Token and If-Match headers
5. Add a JSON body setting the closeoutOfficeId property to a transportation office id in your local database that provides closeout.
6. See that the response includes the closeout office id and nested closeout office data

## Verification Steps for Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.



### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screen Shot 2022-12-30 at 11 39 54 AM](https://user-images.githubusercontent.com/52669884/210102236-0f668a99-f4dc-42c4-841e-1be06e93a19f.png)
![Screen Shot 2022-12-30 at 11 40 26 AM](https://user-images.githubusercontent.com/52669884/210102260-9716056f-79c2-4e75-a20a-1addc8d5d236.png)

![Screen Shot 2022-12-30 at 1 34 01 PM](https://user-images.githubusercontent.com/52669884/210102277-531bb894-721b-41e3-8b55-1243c0db06c3.png)
![Screen Shot 2022-12-30 at 1 34 19 PM](https://user-images.githubusercontent.com/52669884/210102288-d427e7d4-9a63-4f0f-a4eb-58077320c080.png)
